modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsRequestCollection.java

```diff
@@ -0,0 +1,343 @@
+package org.apache.ignite.internal.processors.query.stat;
+
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.cluster.ClusterNode;
+import org.apache.ignite.internal.processors.affinity.AffinityTopologyVersion;
+import org.apache.ignite.internal.processors.cache.CacheGroupContext;
+import org.apache.ignite.internal.processors.query.h2.SchemaManager;
+import org.apache.ignite.internal.processors.query.h2.opt.GridH2Table;
+import org.apache.ignite.internal.processors.query.stat.messages.StatsCollectionRequest;
+import org.apache.ignite.internal.processors.query.stat.messages.StatsCollectionResponse;
+import org.apache.ignite.internal.processors.query.stat.messages.StatsKeyMessage;
+import org.apache.ignite.internal.processors.query.stat.messages.StatsObjectData;
+import org.apache.ignite.internal.util.typedef.F;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Statistics collection routine.
+ */
+public class IgniteStatisticsRequestCollection {
+    /** Current collections, collection id to collection status map. */
+    private final Map<UUID, StatCollectionStatus> currColls = new ConcurrentHashMap<>();
+
+    /** Schema manager. */
+    private final SchemaManager schemaMgr;
+
+    /**
+     * Constructor.
+     *
+     * @param schemaMgr Schema manager.
+     */
+    public IgniteStatisticsRequestCollection(SchemaManager schemaMgr) {
+        this.schemaMgr = schemaMgr;
+    }
+
+    /**
+     * Add new statistics collection status.
+     *
+     * @param status status to add.
+     */
+    public void addCollection(StatCollectionStatus status) {
```

Comment:
Method is declared as to add a collection, but got 'status'.

Reply:
It was about statistics **collection** status. I renamed it.